### DISCUSSION TITLE
Why DAGS removed from UI after dag_run completion after airflow update?

### DISCUSSION THREAD

**Discussion by RubcovMindbox** [apache/airflow#20953] [Q&A] [answered]
I create dags dynamically in loop inside python file located in dags folder using `globals()[task_id] = dag`.
Get `task_id`and other data necessary to create dag from anohter service via HTTP requests. 
In airflow version 2.1.2 these dags remained visible in the UI after dag_run was completed.
But when I update airflow to 2.2.3 this behavior changed:
Dags removed from UI after dag_run finishes.
Could you explain why? I'd like to have completed dags visible in UI

**Comment by potiuk** [comment, marked as answer]
1) it is a bad practice to use HTTP service in your top level code when dynamically generating DAGs - see https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#dynamic-dag-generation for some strategies that are better

2) You need to make sure that your dynamic DAG creation also creates the completed DAGs. Airflow dynamically parses the DAGs every 30 seconds and in 2.2 it will correctly (it was a bug before) remove the DAGs that have disappeared - assuming that the intention is they should be deleted. In your case you should dynamically create all the DAGS wyou want to see in the UI every time the DAG is parsed - now and in the future.

**Reply by RubcovMindbox** [reply]
Thank you a lot! Very helpful.
